spec/lib/prison_day_spec.rb

```diff
@@ -0,0 +1,226 @@
+require 'rails_helper'
+
+RSpec.describe PrisonDay do
+  DAYS = {
+    monday:     Date.parse('Mon 13 July 2015'),
+    tuesday:    Date.parse('Tue 14 July 2015'),
+    wednesday:  Date.parse('Wed 15 July 2015'),
+    thursday:   Date.parse('Thu 16 July 2015'),
+    friday:     Date.parse('Fri 17 July 2015'),
+    saturday:   Date.parse('Sat 18 July 2015'),
+    sunday:     Date.parse('Sun 19 July 2015')
+  }.freeze
+
+  WEEKDAYS = DAYS.except(:saturday, :sunday).freeze
+  WEEKEND_DAYS = DAYS.slice(:saturday, :sunday).freeze
+
+  let(:slots_for_everyday) do
+    {
+      "mon"=>["1400-1600"],
+      "tue"=>["1400-1600"],
+      "wed"=>["1400-1600"],
+      "thu"=>["1400-1600"],
+      "fri"=>["1400-1600"],
+      "sat"=>["0930-1130"],
+      "sun"=>["1400-1600"]
+    }
+  end
+
+  let(:prison_data) do
+    {
+      "nomis_id"=>"RCI",
+      "canned_responses"=>true,
+      "enabled"=>true,
+      "phone"=>"01634 803100",
+      "email"=>"pvb.rochester@maildrop.dsd.io",
+      "instant_booking"=>false,
+      "address"=>["1 Fort Road", "Rochester", "Kent", "ME1 3QS"],
+      "unbookable"=> [],
+      "slot_anomalies"=> {},
+      "slots"=> slots_for_everyday
+    }
+  end
+
+  def prison_from(prison_data)
+    Prison.new 'Example Prison', prison_data
+  end
+
+  describe '::BANK_HOLIDAYS' do
+    it 'contains a duplicate array of the data found in the Rails configuration' do
+      expect(described_class::BANK_HOLIDAYS).to match_array Rails.configuration.bank_holidays
+    end
+  end
+
+  describe '.staff_working_day?' do
+    context 'on a regular weekday' do
+      WEEKDAYS.each do |day_name, date|
+        subject { described_class.new(date, prison_from(prison_data)) }
+        context "like a #{day_name}" do
+          specify { expect(subject.staff_working_day?).to be true }
+        end
+      end
+    end
+
+    context 'on a public holiday' do
+      let(:bank_holidays) { [Date.new(2015, 8, 31)] }
+      before { stub_const("#{described_class.name}::BANK_HOLIDAYS", bank_holidays) }
+      subject { described_class.new(bank_holidays.first, prison_from(prison_data)) }
+
+      specify { expect(subject.staff_working_day?).to be false }
```

Comment:
Going to suggest we keep the contexts together for the time being so the spec doesn't end up scattered and hard to reason about. However, as we progress perhaps we'll introduce factories for generating different prisons with specific characteristics and we can tidy this up a bit. Something to put on the radar anyways.